### PR TITLE
HOSTEDCP-969: Only track available metric once

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
/hold
needs https://github.com/openshift/hypershift/pull/2477

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by sjenning** [comment]
needs rebase but https://github.com/openshift/hypershift/pull/2477 merged
/hold cancel

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -640,6 +643,19 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	// availability in the ultimate HostedCluster Available condition)
 	{
 		meta.SetStatusCondition(&hcluster.Status.Conditions, computeHostedClusterAvailability(hcluster, hcp))
+
+		// SLI: Hosted Cluster available duration.
+		availableTime := clusterAvailableTime(hcluster)
+		if availableTime != nil {
+			hostedClusterAvailableDuration.WithLabelValues(hcluster.Name).Set(*availableTime)
+		}
+		setHasBeenAvailable(hcluster)
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go]: makes sense, I dropped the function and make the call inline.
We can later on follow up to abstract and have the whole logic e.g. reportAvailability to set the condition, the metric and the annotation and unit test that.

**Review by jparrill — COMMENTED** [review]
/lgtm

**Pull request merged**